ssh/tailssh/tailssh.go

```diff
@@ -1110,9 +1110,13 @@ var (
 	errRuleExpired    = errors.New("rule expired")
 	errPrincipalMatch = errors.New("principal didn't match")
 	errUserMatch      = errors.New("user didn't match")
+	errInvalidConn    = errors.New("invalid connection state")
 )
 
 func (c *conn) matchRule(r *tailcfg.SSHRule, pubKey gossh.PublicKey) (a *tailcfg.SSHAction, localUser string, err error) {
+	if c == nil || c.info == nil {
+		return nil, "", errInvalidConn
+	}
```

Comment:
I don't think we want checkStillValid (via OnPolicyChange) to boot connections that are just connecting in at that moment.

If they haven't gone though the auth phase yet (they only TCP connected but didn't send auth type none yet) then they'll get rejected later at that phase. We can just skip over them in OnPolicyChange and not worry about booting them just because they got unlucky with timing.